db/compaction/compaction.cc

```diff
@@ -238,7 +238,11 @@ Compaction::Compaction(
       inputs_(PopulateWithAtomicBoundaries(vstorage, std::move(_inputs))),
       grandparents_(std::move(_grandparents)),
       score_(_score),
-      bottommost_level_(IsBottommostLevel(output_level_, vstorage, inputs_)),
+      bottommost_level_(
+          (_compaction_reason == CompactionReason::kExternalSstIngestion ||
+           _compaction_reason == CompactionReason::kRefitLevel)
+              ? false
+              : IsBottommostLevel(output_level_, vstorage, inputs_)),
```

Comment:
[Note] Logic in `IsBottommostLevel()` does not consider cases like RefitLevel() and file ingestion. So I explicitly excluded them to avoid touching its logic.

Reply:
The original logic is correct to keep - it's just that the bottommost concept is not applied to RefitLevel (where we can output to L0 when L0 is empty) nor file ingestion (where the input files do not exists in LSM prior to the "file-ingestion compaction). I also  didn't want to relax the logic in `IsBottommostLevel` for `RefitLevel` or file ingestion cuz they are really here for conflict checking, not for anything else. Updated `CompactionReason::kExternalSstIngestion/kRefitLevel` comments.